src/commander/option.cr

```diff
@@ -0,0 +1,32 @@
+class Commander::Option(T)
+  getter value, flag
+
+  def initialize(@value : T, @flag : Flag); end
+
+  def key
+    flag.name
+  end
+
+  def global?
+    !!flag.global
+  end
+
+  def self.build(value, flag)
```

Comment:
I've added this helper method as using `Option.new` didn't correctly recognize the type. I was getting a following error:

```
Error: no overload matches 'Commander::Options#set' with types String, Commander::Option(Bool | Float32 | Float64 | Int32 | Int64 | String | Nil)
```